pkg/jobs/registry.go

```diff
@@ -1458,3 +1462,24 @@ func (r *Registry) maybeRecordExecutionFailure(ctx context.Context, err error, j
 		log.Warningf(ctx, "failed to record error for job %d: %v: %v", j.ID(), err, err)
 	}
 }
+
+// CheckPausepoint returns a PauseRequestError if the named pause-point is
+// set.
+//
+// This can be called in the middle of some job implementation to effectively
+// define a 'breakpoint' which, when reached, will cause the job to pause. This
+// can be very useful in allowing inspection of the persisted job state at that
+// point, without worrying about catching it before the job progresses further
+// or completes. These pause points can be set or removed at runtime.
+func (r *Registry) CheckPausepoint(name string) error {
+	s := debugPausepoints.Get(&r.settings.SV)
+	if s == "" {
+		return nil
+	}
+	for _, point := range strings.Split(s, ",") {
+		if name == point {
+			return MarkPauseRequestError(errors.Newf("pause point hit: %s", name))
```

Comment:
If you want, you can add something like
```
InfofDepth(ctx, 1, "puasing job due to %q pausepoint", point)
```

Reply:
Eh, but we're not actually pausing here; we're just discovering that a pause point is enabled, and handing an error indicating that fact to the caller - what they do with it is their decision, and theirs to log too.